src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -61,6 +61,15 @@ public static Connection Connect(string host, int port, CancellationToken cancel
                     socket.Connect(new DnsEndPoint(host, port));
                 }
             }
+            catch (SocketException se)
+            {
+                socket.Dispose();
+
+                // SocketConnectionFactory wraps SocketException in NetworkException. Do the same here.
+                NetworkException ne = NetworkErrorHelper.MapSocketException(se);
+
+                throw CreateWrappedException(ne, host, port, cancellationToken);
+            }
```

Comment:
Wouldn't it be wise to add a test for this path?

Reply:
Probably, yeah. I want to get this in so people aren't seeing this in CI.